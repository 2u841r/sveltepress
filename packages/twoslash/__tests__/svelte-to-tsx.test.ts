import { readFileSync } from 'node:fs'
import { resolve } from 'node:path'
import { describe, expect, it } from 'vitest'
import { svelte2tsx } from 'svelte2tsx'

const svelteCode = readFileSync(resolve(import.meta.dirname, 'test.svelte'), 'utf-8')

describe('shiki', () => {
  it('twoslash svelte', async () => {
    const { code } = svelte2tsx(svelteCode)
    expect(code).toMatchSnapshot()
  })
})
