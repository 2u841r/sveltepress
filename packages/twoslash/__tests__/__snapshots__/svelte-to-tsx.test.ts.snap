// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shiki > twoslash svelte 1`] = `
"///<reference types="svelte" />
;
  export const FOO = 'BAR'
;;

import { onMount } from 'svelte'
function render() {

  

  let count = 0

  onMount(() => {
    console.log('mounted')
  })
;
async () => {



 { svelteHTML.createElement("button", {  "on:click":() => count++,});
    count;
 }
};
return { props: /** @type {Record<string, never>} */ ({}), slots: {}, events: {} }}

export default class extends __sveltets_2_createSvelte2TsxComponent(__sveltets_2_partial(__sveltets_2_with_any_event(render()))) {
}"
`;
